src/Microsoft.Framework.BufferEntryCollection.Internal/BufferEntryCollection.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.Framework.Internal
+{
+    /// <summary>
+    /// An ordered collection of strings with an enumerator that iterates over it as a sequence.
+    /// </summary>
+    internal class BufferEntryCollection : IEnumerable<string>
+    {
+        // Specifies the maximum size we'll allow for direct conversion from
+        // char arrays to string.
+        private const int MaxCharToStringLength = 1024;
+        private readonly List<object> _buffer = new List<object>();
+        private bool _isModified = false;
+
+        /// <summary>
+        /// Holds the collection of strings.
+        /// </summary>
+        public IReadOnlyList<object> BufferEntries
```

Comment:
why is this exposed in addition to the enumerator?  exposing this implementation detail (if actually required) indicates the API isn't complete enough